src/status.rs

```diff
@@ -126,7 +131,10 @@ impl StatusCode {
     /// ```
     #[inline]
     pub fn as_str(&self) -> &str {
-        CODES_AS_STR[(self.0 - 100) as usize]
+        let offset = (self.0.get() - 100) as usize;
+        let offset = offset * 3;
+
+        unsafe { CODE_DIGITS.get_unchecked(offset..offset+3) }
```

Comment:
Curious, is there a reason to favor an unsafe get here instead of just indexing? I realize the code is definitely correct, but also appreciate the point justifying `unsafe` with "it's significantly faster" or "it's impossible otherwise". Not a blocker, just something I think is worth being explicit about.

Reply:
Added e93c972 to enable debug asserted indexing in debug mode and to document the invariants which make `get_unchecked` safe in release mode.  Since the range is checked and `CODE_DIGITS` is `const` of known format, it seems reasonable to skip the UTF-8 character boundary checks, at least in release mode.